applications/CoSimulationApplication/tests/test_cosim_EMPIRE_API.py

```diff
@@ -0,0 +1,290 @@
+        KM.Logger.GetDefaultOutput().SetSeverity(KM.Logger.Severity.WARNING) # mute MP-IO
+
+        model_part_io = KM.ModelPartIO(GetFilePath("generic_mdpa_files/Mok_CFD"))
+        model_part_io.ReadModelPart(model_part)
+
+        KM.Logger.GetDefaultOutput().SetSeverity(severity)
+
+        KratosCoSim.EMPIRE_API.EMPIRE_API_sendMesh(model_part)
+
+        # can directly use filecmp because there are no decimal-number issues
+        self.assertTrue(filecmp.cmp(GetFilePath("reference_files/EMPIRE_mesh_For_Sending.vtk_ref"), "EMPIRE_mesh_For_Sending.vtk"))
+
+    def test_EMPIRE_API_recvMesh(self):
+        mp_name = "For_Receiving"
+        mesh_file_name = GetMeshFileName(mp_name)
+        model = KM.Model()
+        model_part = model.CreateModelPart(mp_name)
+        model_part_ref = model.CreateModelPart("For_Checking")
+
+        copyfile("reference_files/EMPIRE_mesh_For_Sending.vtk_ref", mesh_file_name)
+
+        KratosCoSim.EMPIRE_API.EMPIRE_API_recvMesh(model_part)
+
+        # make sure that the file was deleted
+        self.assertFalse(os.path.isfile(mesh_file_name))
+
+        # reading reference ModelPart (with which the ref-vtk was created)
+        severity = KM.Logger.GetDefaultOutput().GetSeverity()
+        KM.Logger.GetDefaultOutput().SetSeverity(KM.Logger.Severity.WARNING) # mute MP-IO
+        model_part_io = KM.ModelPartIO(GetFilePath("generic_mdpa_files/Mok_CFD"))
+        model_part_io.ReadModelPart(model_part_ref)
+        KM.Logger.GetDefaultOutput().SetSeverity(severity)
+
+        self.assertEqual(model_part.NumberOfNodes(), model_part_ref.NumberOfNodes())
+        self.assertEqual(model_part.NumberOfElements(), model_part_ref.NumberOfElements())
+
+        self.__CompareNodes(model_part.Nodes, model_part_ref.Nodes)
+
+        for elem, elem_ref in zip(model_part.Elements, model_part_ref.Elements):
+            self.assertEqual(elem.Id, elem_ref.Id)
+            self.__CompareNodes(elem.GetNodes(), elem_ref.GetNodes())
+
+    def test_EMPIRE_API_sendDataField(self):
+        fct_to_test = KratosCoSim.EMPIRE_API.EMPIRE_API_sendDataField
+        file_name_fct_ptr = GetDataFieldFileName
+
+        array = [1.0, 2.5, 3.5, -99.11, -0.02, 555.5, -99.114, 0.02, 565.5, 10.0, 78.44]
+
+        self.__TestArraySend(fct_to_test, file_name_fct_ptr, array)
+
+    def test_EMPIRE_API_recvDataField(self):
+        fct_to_test = KratosCoSim.EMPIRE_API.EMPIRE_API_recvDataField
+        file_name_fct_ptr = GetDataFieldFileName
+
+        array = [1.0, 2.5, 3.5, -99.11, -0.02, 555.5, -99.114, -99.11, -0.02, 555.5, 0.02, 565.5, 10.0, 78.44]
+
+        self.__TestArrayReceive(fct_to_test, file_name_fct_ptr, array)
+
+    def test_EMPIRE_API_sendSignal_double(self):
+        fct_to_test = KratosCoSim.EMPIRE_API.EMPIRE_API_sendSignal_double
+        file_name_fct_ptr = GetSignalFileName
+
+        array = [1.0, 2.5, 3.5, -99.11, -0.02, 555.5, -99.114, 0.02, 565.5, 10.0, 78.44]
```

Comment:
Is this required ? The above test also tests the same !

Reply:
the test above tests `EMPIRE_API_sendDataField`, this test is for `EMPIRE_API_sendSignal_double`
I just set it up to use the same testing framework

Reply:
ah ok .. i missed line 128 !